src/langrunner.rs

```diff
@@ -205,28 +216,40 @@ impl LangRunnerProcess {
         // Spawn a thread to handle algorithm stderr - we do this here so we shouldn't get get stuck
         // waiting for stuff to be read from stderr when we are loading the algorithm
         let arc_request_id_err = request_id.clone();
+        let arc_stderr_buf = stderr_buf.clone();
         thread::spawn(move || {
             let reader = BufReader::new(stderr);
             for line_result in reader.lines() {
                 let req_id = arc_request_id_err
-                    .read().expect("failed to get read handle on request_id for stdout reading")
+                    .read().expect("failed to get read handle on request_id for stderr reading")
                     .clone().unwrap_or_else(|| "-".to_owned());
                 match line_result {
-                    Ok(line) => info!("{} {} {}", "ALGOERR", req_id, line),
+                    Ok(line) => {
+                        match arc_stderr_buf.lock() {
+                            Ok(mut lines) => {
+                                lines.push_str(&line);
+                                lines.push('\n');
+                            }
+						    Err(err) => error!("{} {} Failed to get lock on stderr buffer: {}", LOG_IDENTIFIER, req_id, err),
```

Comment:
nit: indentaton looks off cause of tabs

Reply:
suggest running `cargo fmt` on in a separate commit or PR - designed to keep formatting consistent without thinking about it, but it will probably touch a lot of lines the first time it's run.

Reply:
we can run `cargo fmt` separately to help clarify these issues, not sure how that ended up happening here.

Reply:
that was me editting in `vim` without a vimprofile to use spaces instead of tabs... :-(